.github/workflows/ElastixGitHubActions.yml

```diff
@@ -154,5 +154,20 @@ jobs:
       with: 
         name: "${{ matrix.os }}"
         path: uploads
+        
+    - name: Login to Docker Hub
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags') && matrix.os == 'ubuntu-18.04'
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+    - name: Build and push
+      if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags') && matrix.os == 'ubuntu-18.04'     
+      run: |
+        mkdir output
+        docker build -t superelastix/elastix:${GITHUB_REF:10} .
+        docker run -u $UID:$GROUPS --mount type=bind,source="$(pwd)"/output,target=/out -v "$(pwd)"/Elastix-source/Testing/Data:/elastix/ superelastix/elastix:${GITHUB_REF:10} elastix -out /out/ -p /elastix/parameters.3D.NC.euler.ASGD.001.txt -f /elastix/3DCT_lung_baseline.mha -m /elastix/3DCT_lung_followup.mha
```

Comment:
Just for my understanding, it runs a registration here, just to test if it works?

Reply:
it seems so to me as well, and then my question would be: is there a check somewhere that this run was successful?

Reply:
Yes, it runs registration to test the docker image.
I assume here that if it runs and doesn't give an error, the docker image is build successfully.
Since the elastix binary is already extensively tested in this workflow.
Or do we want more test to verify the docker image as well?

Reply:
I don't think more extensive testing is needed, it's enough indeed to see if it runs and doesn't give an error, but I didn't see an explicit check here whether this was the case (for example checking for the return value of 'docker run'). if the script automatically stops when 'docker run' gives an error then this also might be fine, but I don't know how this works with this yml files.

Reply:
The script indeed stops when 'docker run' gives an error. It's a bit hard to see this on this PR though, since the script includes a check to see if the PR/commit is tagged before it runs the docker part of the workflow. Since this PR is not tagged, the CI will not run the docker part.
The CI on [this fork](https://github.com/ViktorvdValk/elastix/runs/1878956471?check_suite_focus=true) shows the workflow run as it will be when a commit is tagged.

Reply:
then it's fine I think!